app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt

```diff
@@ -11,12 +12,24 @@ import com.example.myweatherapp.models.SourceModel
 import com.example.myweatherapp.viewHolders.NewsHolder
 import com.example.myweatherapp.viewHolders.SourceHolder
 
-open class NewsAdapter(private val newsListener: OnItemClickListener, private val sourceListener: OnSourceClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+open class NewsAdapter(private val newsListener: OnItemClickListener, private val sourceListener : OnSourceClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
-    var newsList = emptyList<IListItem>()
+    var newsList = mutableListOf<IListItem>()
+    var favoriteUrlList = mutableListOf<String>()
```

Comment:
Why do you use two lists in an adapter again? It's wrong and you'd better to use only one list here

Reply:
left only one list in the adapter